src/dateparser/dateparser.js

```diff
@@ -125,7 +125,7 @@ angular.module('ui.bootstrap.dateparser', [])
 
     if ( results && results.length ) {
       var fields, dt;
-      if (baseDate) {
+      if (baseDate && Object.prototype.toString.call(baseDate) === "[object Date]" && !isNaN(baseDate.getTime())) {
```

Comment:
Why not use `angular.isDate` to check if it is a date instead of the `toString` call?